### ISSUE TITLE
Failed install using Nix

### ISSUE THREAD

**Issue by JaMaeh** [hacspec/hax#503]
Hey there,

I am trying to install hax on my WSL 2.0 using Nix and it doesn't seem to work.

It stops after doing 98% of the jobs throwing a ton of alerts and warnings. I don't see an error as such, I just get a 

> Command got signal KILL.

 

This is the terminal output:

```
error: builder for '/nix/store/kj6nnr7jijriscwcagfxq24knkfwbs9h-ocaml4.14.1-hax-engine-0.0.1.drv' failed with exit code 1;
       last 10 log lines:
       > Alert deprecated: Base.print_endline
       > [2016-09] this element comes from the stdlib distributed with OCaml.
       > Use [Stdio.print_endline] instead.
       > File "bin/lib.ml", line 115, characters 6-35:
       > 115 |       Printexc.raise_with_backtrace exn bt
       >             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       > Alert deprecated: module Base.Printexc
       > [2016-09] this element comes from the stdlib distributed with OCaml.
       > Use [Exn] or [Backtrace] instead.
       >
       For full logs, run 'nix log /nix/store/kj6nnr7jijriscwcagfxq24knkfwbs9h-ocaml4.14.1-hax-engine-0.0.1.drv'.
error: 1 dependencies of derivation '/nix/store/39iprgmipsf4n9xnc3l1zajzq5bh9kaj-hax-0.0.1.drv' failed to build
```

I am happy to share more from the logs if needed. 

Thanks for your help in advance!
Jannik

**Comment by W95Psp** [comment]
Hi,

Mmh, sadly, that's a memory issue...
The OCaml build gets killed after a while because the compiler is running out of memory, see the issue https://github.com/hacspec/hax/issues/197.

That is something very problematic, and we hope to improve the OCaml build so that it doesn't require such a huge memory (around 20/30GB). However, we won't solve that today!

Right now, the easiest patch for you is to use the binary caches: if you already have `cachix` installed, just run `cachix use hax`.

I can also tweak the Nix expressions to set settings based on hardware constraints, but that's not great...
How much memory do you have on your computer?

**Comment by JaMaeh** [comment]
Hey, that fixed the problem. Thank you very much!

**Comment by W95Psp** [comment]
Nice, I'm happy it worked!

Just to let you know, I actually re-looked into memory issues (https://github.com/hacspec/hax/issues/197) and solved that yesterday (PR #504). Thus, now even without binary cache you should be able to build hax without any issue.
